common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -316,6 +388,19 @@ static ByteBuffer allocateDirectNoCleaner(int capacity) {
         return newDirectBuffer(UNSAFE.allocateMemory(capacity), capacity);
     }
 
+    static byte[] allocateUninitializedArray(int size) {
+        if (ALLOCATE_ARRAY_METHOD == null) {
+            return new byte[size];
+        }
+        try {
+            return (byte[]) ALLOCATE_ARRAY_METHOD.invoke(INTERNAL_UNSAFE, byte.class, size);
```

Comment:
Do we build with JDK 8 now? Then we could delegate this into a separate class that uses `MethodHandle`?